src/hazelcore/Builtins.re

```diff
@@ -1,6 +1,46 @@
+module Impls = {
+  open EvaluatorResult;
+
+  let int_of_float = (ident, args, evaluate) =>
```

Comment:
It might be easier to define a functor around the entire Builtins module rather than taking in the evaluator in each function.

Reply:
How can we create a functor for a file, as opposed to a nested module inside the file module?

Reply:
`evaluate` isn't really relevant to other places that use `Builtins` (such as `Contexts.re`, which uses `Builtins.ctx`), so I think that a functor is perhaps not the best approach.

Reply:
You can define an FBuiltins.F functor that takes the evaluator as input, then define Builtins.re as `include FBuiltins(Evaluator)` and use Builtins normally from there. That breaks up the circularity.

Reply:
Hmm, I tried some different things but couldn't figure out how to do what you mean and avoid a circular dependency. If Builtins has `include FBuiltins(Evaluator)`, and then Evaluator calls Builtins, there would still be a circular dependency?